### ISSUE TITLE
Inconsistent behavior when navigating after a delay vs navigating without delay

### ISSUE THREAD

**Issue by KhatibFX** [GIfatahTH/states_rebuilder#85]
Hey @GIfatahTH , sorry for bothering you again! So after the newest update, I no longer get the previous issue where the model is already injected when navigating instantly, but there is another issue. Here is the minimal reproducing code:

```dart
import 'package:flutter/material.dart';
import 'package:states_rebuilder/states_rebuilder.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: FirstRoute(),
    );
  }
}

class FirstRoute extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('First Route'),
      ),
      body: Center(
        child: RaisedButton(
          child: Text('Open route'),
          onPressed: () {
            // Navigate to second route when tapped.
            Navigator.push(
              context,
              MaterialPageRoute(builder: (context) => SecondRoute()),
            );
          },
        ),
      ),
    );
  }
}

class SecondRoute extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Injector(
        inject: [
          Inject<Example>(() => Example()),
        ],
        builder: (BuildContext context) {
          return Scaffold(
            appBar: AppBar(
              title: StateBuilder(
                observe: () => RM.get<Example>(),
                builder: (_, ReactiveModel<Example> exampleRM) {
                  return Text(exampleRM.state.title);
                },
              ),
            ),
            body: Center(
              child: RaisedButton(
                onPressed: () {
                  RM
                      .get<Example>()
                      .setState((state) => state.title = 'Title Two');
                  // Navigate back to first route when tapped.
                  Navigator.pop(context);
                },
                child: Text('Go back!'),
              ),
            ),
          );
        });
  }
}

class Example {
  String title = 'Title one';
}

```

I only made a few changes to the previous example. Now the Example class has a title, and the second page gets its title from that.

If you click on open route and then go back with some delay and again open route, the second page's title will always be 'Title one' at the beginning. But if you do the same action without delay, the second page's title will always be 'Title Two'.

This is affecting me on a larger scale than this example, as I have multiple asynchronous calls that change state one after another, but when I navigate quickly, the screen will show the last asynchronous call state that managed to execute first, until the first asynchronous call changes state successfully.

I hope my description wasn't vague, and again thanks for your marvelous effort with this library!

**Comment by GIfatahTH** [comment]
@KhatibFX 
Thank for your contribution. I really appreciate your issues. 
Your issues never bother me, on the contrary they make me happy because they encourage me, and because of your minimal reproducing code you provide:

To understand the issue:
Injector adds an instance of Example class to the container (service locator) in its initState and removes it in the dispose hook.

What happens when you pop back to the first page the dispose of the second page is called and the model is removed and when you push again to the second page the initState is called and the model is added.

When you pop push quickly (more rapid than animation transition) the old second page is not disposed yet but the initState of the new second page is called. As this is the case when RM.get<Example>().state is called it gets the state of the old second page that is not disposed yet.

This what makes the inconstancy between slow and rapid navigation.

To solve:
In injector.dart file and line 289 please do the following changes:
remove :
```dart
allRegisteredModelInApp[name].add(lastInject.first);
```
and add
```dart
allRegisteredModelInApp[name].add(inject);
```


Please try that and feed me back.
Thanks again for your contribution.

**Comment by GIfatahTH** [comment]
Issue solved in v2.0.0.
Do not hesitate to reopen it if the problem persists.
